### DISCUSSION TITLE
Wyze v2 ethernet not working

### DISCUSSION THREAD

**Discussion by clee604** [gtxaspec/wz_mini_hacks#94] [Q&A] [answered]
Not sure if I should create an issue for this or what other debugging steps I can take first but I have all 3 ethernet types (asix,r8152,cdc_ether) and I can't seem to get it to work on my wyze v2. Wifi works if I set the ethernet flag to false, but with it true and with the driver set it does not work. 

Using the exact same cable and configuration file, all three adapters work on my v3 camera as long as I have the corresponding driver set correctly.

Is there anything else I can try to get the ethernet working on the v2?

**Comment by gtxaspec** [comment]
boot the camera with the Ethernet adapter, wait a couple of minutes,power the camera down and take out the sd card, grab the log file "wz_user.log" and take a look.  otherwise you can share it here, remove any passwords before hand.

**Comment by clee604** [comment]
I compared the log between the v2 and the v3 and this snippet is the main difference. It's not very descriptive besides the SIOCGIFFLAGS: No such device error if that means anything to you.
+ rename_interface eth0
+ echo 'renaming interfaces'
renaming interfaces
+ ifconfig eth0 down
ifconfig: SIOCGIFFLAGS: No such device
+ ifconfig wlan0 down
+ /opt/wz_mini/bin/busybox ip link set wlan0 name wlanold
+ /opt/wz_mini/bin/busybox ip addr flush dev wlanold
+ /opt/wz_mini/bin/busybox ip link set eth0 name wlan0
ip: SIOCSIFNAME: No such device
+ eth_wlan_up
+ ifconfig wlan0 up
ifconfig: SIOCGIFFLAGS: No such device
+ pkill udhcpc
+ udhcpc -i wlan0 -x hostname:WCV2-LivingFloor -p /var/run/udhcpc.pid -b
udhcpc: SIOCGIFINDEX: No such device
+ '[[' true '==' true ]]
+ mount -o bind /opt/wz_mini/bin/wpa_cli.sh /system/bin/wpa_cli
+ break

**Comment by gtxaspec** [comment]
ok what about "dmesg.log" it should show the ethernet adapter being loaded. check that too

**Comment by gtxaspec** [comment]
it looks like either the modules are not loading properly, or the Ethernet device is named something other than eth0... dmesg should give more insight.

**Comment by clee604** [comment]
Here is the dmesg from the time it looks like it starts network stuff:

   11.163022] RTL871X: module init start
[   11.166920] RTL871X: rtl8189ftv v4.3.24.7_21113.20170208.nova.1.02
[   11.174408] RTL871X: build time: May 26 2020 12:44:49
[   11.179655] wlan power on
[   11.202062] RTL871X: module init ret=0
[   11.247474] mmc1: card claims to support voltages below the defined range. These will be ignored.
[   11.272383] mmc1: new SDIO card at address 0001
[   11.279824] RTL871X: ++++++++rtw_drv_init: vendor=0x024c device=0xf179 class=0x07
[   11.342860] RTL871X: HW EFUSE
[   11.345966] RTL871X: hal_com_config_channel_plan chplan:0x20
[   11.479286] RTL871X: rtw_regsty_chk_target_tx_power_valid return _FALSE for band:0, path:0, rs:0, t:-1
[   11.490506] RTL871X: rtw_ndev_init(wlan0) if1 mac_addr=c8:02:8f:8c:f7:bd
[   12.925673] jffs2: warning: (99) jffs2_do_read_inode_internal: Truncating ino #66 to 336704 bytes failed because it only had 20458 bytes to start with!
[   19.120046] name : i2c-gpio1 nr : 1
[   19.126446] name : i2c0 nr : 0
[   19.190564]  sensor_read: addr=0xa value = 0xf
[   19.195646]  sensor_read: addr=0xb value = 0x23
[   19.200350] info: success sensor find : jxf23
[   19.204914] misc sinfo_release
[   19.862241] set sensor gpio as PA-low-10bit
[   19.866589] probe ok ------->jxf23
[   20.018592] -----jxf23_detect: 631 ret = 0, v = 0x0f
[   20.019005] -----jxf23_detect: 639 ret = 0, v = 0x23
[   20.019019] jxf23 0-0040: jxf23 chip found @ 0x40 (i2c0)
[   20.024557] tx_isp: Registered sensor subdevice jxf23 0-0040
[   20.030698] do not support ctrl->cmd ====1
[   20.030720] do not support ctrl->cmd ====0
[   20.034341] do not support ctrl->cmd ====1
[   20.309375] do not support ctrl->cmd ====1
[   20.310288] jxf23 stream on
[   20.521330] ###### image_tuning_v4l2_open 4219 #######
[   20.536699] before register 0x1f value : 0x01
[   20.537023] after register 0x1f value : 0x81
[   20.537038] do not support ctrl->cmd ====1
[   20.817635] &&& chan1  scaler.max_width = 1920 max_height = 1080  min_width = 128 min_height = 128 &&&
[   21.097595] &&& chan2  scaler.max_width = 800 max_height = 800  min_width = 128 min_height = 128 &&&
[   21.120490] codec_codec_ctl: set sample rate...
[   21.120955] codec_codec_ctl: set device...
[   21.513169] codec_set_device: set device: MIC...
[   21.729807] codec_codec_ctl: set repaly channel...
[   21.729831] codec_codec_ctl: set sample rate...
[   21.730073] codec_codec_ctl: set device...
[   21.961654] codec_set_device: set device: speaker...
[   23.407608] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
[   24.728952] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
[   37.340205] RTL871X: rtw_set_802_11_connect(wlan0)  fw_state=0x00000008
[   37.477764] RTL871X: start auth
[   37.484356] RTL871X: auth success, start assoc
[   37.494098] RTL871X: assoc success
[   37.500162] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
[   37.509770] RTL871X: recv eapol packet
[   37.525626] RTL871X: send eapol packet
[   37.584751] RTL871X: recv eapol packet
[   38.394708] RTL871X: send eapol packet
[   38.412024] RTL871X: set pairwise key camid:4, addr:80:2a:a8:17:5f:ce, kid:0, type:AES
[   38.437363] RTL871X: set group key camid:5, addr:80:2a:a8:17:5f:ce, kid:2, type:AES
[   45.066506] usbcore: registered new interface driver cdc_ether
[   45.283688] RTL871X: OnDeAuth(wlan0) reason=6, ta=80:2a:a8:17:5f:ce, ignore=0
[   45.309479] cfg80211: Calling CRDA to update world regulatory domain

**Comment by clee604** [comment]
Comparing it to when the ethernet flag is false, I pulled the dmesg (it is from a slightly older version of the wz_mini, not sure if this makes a difference) I don't see this CRDA call in the log. This is pretty much all glibberish to me though, hopefully it makes more sense to you.

[   15.824491] RTL871X: module init start
[   15.828387] RTL871X: rtl8189ftv v4.3.24.7_21113.20170208.nova.1.02
[   15.836007] RTL871X: build time: May 26 2020 12:44:49
[   15.841279] wlan power on
[   15.862549] RTL871X: module init ret=0
[   15.907521] mmc1: card claims to support voltages below the defined range. These will be ignored.
[   15.930835] mmc1: new SDIO card at address 0001
[   15.939013] RTL871X: ++++++++rtw_drv_init: vendor=0x024c device=0xf179 class=0x07
[   16.000327] RTL871X: HW EFUSE
[   16.003491] RTL871X: hal_com_config_channel_plan chplan:0x20
[   16.138050] RTL871X: rtw_regsty_chk_target_tx_power_valid return _FALSE for band:0, path:0, rs:0, t:-1
[   16.150537] RTL871X: rtw_ndev_init(wlan0) if1 mac_addr=c8:02:8f:8c:f7:bd
[   19.563353] jffs2: warning: (108) jffs2_do_read_inode_internal: Truncating ino #66 to 336704 bytes failed because it only had 20458 bytes to start with!
[   24.094272] name : i2c-gpio1 nr : 1
[   24.097895] name : i2c0 nr : 0
[   24.166526]  sensor_read: addr=0xa value = 0xf
[   24.171703]  sensor_read: addr=0xb value = 0x23
[   24.176409] info: success sensor find : jxf23
[   24.180950] misc sinfo_release
[   25.156072] set sensor gpio as PA-low-10bit
[   25.160422] probe ok ------->jxf23
[   25.312308] -----jxf23_detect: 631 ret = 0, v = 0x0f
[   25.312725] -----jxf23_detect: 639 ret = 0, v = 0x23
[   25.312739] jxf23 0-0040: jxf23 chip found @ 0x40 (i2c0)
[   25.318303] tx_isp: Registered sensor subdevice jxf23 0-0040
[   25.324307] do not support ctrl->cmd ====1
[   25.324327] do not support ctrl->cmd ====0
[   25.327909] do not support ctrl->cmd ====1
[   25.602898] do not support ctrl->cmd ====1
[   25.603811] jxf23 stream on
[   25.815787] ###### image_tuning_v4l2_open 4219 #######
[   25.831358] before register 0x1f value : 0x01
[   25.831815] after register 0x1f value : 0x81
[   25.831830] do not support ctrl->cmd ====1
[   26.110768] &&& chan1  scaler.max_width = 1920 max_height = 1080  min_width = 128 min_height = 128 &&&
[   26.389271] &&& chan2  scaler.max_width = 800 max_height = 800  min_width = 128 min_height = 128 &&&
[   26.404068] codec_codec_ctl: set sample rate...
[   26.404568] codec_codec_ctl: set device...
[   26.801689] codec_set_device: set device: MIC...
[   27.002285] codec_codec_ctl: set repaly channel...
[   27.002313] codec_codec_ctl: set sample rate...
[   27.002558] codec_codec_ctl: set device...
[   27.241715] codec_set_device: set device: speaker...
[   28.817001] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
[   30.101846] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
[   39.960151] RTL871X: rtw_set_802_11_connect(wlan0)  fw_state=0x00000008
[   40.103047] RTL871X: start auth
[   40.110182] RTL871X: auth success, start assoc
[   40.121022] RTL871X: assoc success
[   40.124824] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
[   40.133372] RTL871X: recv eapol packet
[   40.145402] RTL871X: send eapol packet
[   40.191196] RTL871X: recv eapol packet
[   40.196000] RTL871X: send eapol packet
[   40.210592] RTL871X: set pairwise key camid:4, addr:80:2a:a8:17:5f:ce, kid:0, type:AES
[   40.240442] RTL871X: set group key camid:5, addr:80:2a:a8:17:5f:ce, kid:1, type:AES
[   46.153256] Adding 131068k swap on /media/mmc/wz_mini/swap.  Priority:-1 extents:1 across:131068k SS

**Comment by gtxaspec** [comment]
check wz_user.log, can you share the lines that start with 'insmod', also can you share the Ethernet module lines from your wz_mini.conf

**Comment by clee604** [comment]
User log:
+ insmod /opt/wz_mini/lib/modules/3.10.14/kernel/drivers/net/usb/usbnet.ko
+ echo cdc_ether
+ tr , '\n'
+ insmod /opt/wz_mini/lib/modules/3.10.14/kernel/drivers/net/usb/cdc_ether.ko

Config:
ENABLE_USB_ETH="true"
ENABLE_USB_ETH_MODULES="cdc_ether"

**Comment by gtxaspec** [comment]
what brand and model ethernet adapter are you using?  If you can ssh to the camera, plug in the usb adapter, and type `lsusb` and you can see if it detects the adapter.

**Comment by clee604** [comment]
I actually tried three different ethernet adapters, which happen to be one of each
- Lenovo usb 2.0 is ASIX
- Aliexpress usb 2.0 is realtek
- Aliexpress usb.20 is cdc

I initially tried these three using my wyze v3 camera and all worked as long as I set the right ethernet driver. None of them work with my wyze v2 camera so I suspect it's not specific to a type of network adapter.

I can SSH in when it's connected to wifi (ie. ethernet is set to false), should I connect the usb ethernet to the wyze v2 but keep ethernet false to get the lsusb?

**Reply by gtxaspec** [reply]
thats correct, keep the wifi on, connect via ssh, plug in your ethernet adapter, then `lsusb`

**Comment by clee604** [comment]
So i remade the SD card with the latest version and now the ethernet is half working (at least I think it is, it is using same ip address as wifi before). I can see the camera in wyze and I can SSH into it but RTSP does not work. The exact same RTSP configuration worked with wifi. 

lsusb gives me the following on a cdc_ether adapter
[root@Ingenic-uc1_1:~]# lsusb
Bus 001 Device 001: ID 1d6b:0002

**Comment by gtxaspec** [comment]
Looks like your USB adapter is not being detected with that kernel module.  You can try another module, or a different adapter.  That lsusb entry is the camera itself.  

Make sure `ENABLE_USB_RNDIS` and `ENABLE_USB_DIRECT` are set to false as well.

**Comment by clee604** [comment]
I moved the same adapter and settings over to the wyze v3 and it works. Here is the lsusb

[root@WCV3-Balcony:~]# lsusb
Bus 001 Device 002: ID 0fe6:9900
Bus 001 Device 001: ID 1d6b:0002

I noticed the v2 didn't update the hostname either when I ssh into the camera, I had set a custom one. Is there something different between the v2 and v3 in the kernel it uses?

**Comment by gtxaspec** [comment]
on the v3, with the adapter plugged in, can you ssh in and type `lsmod`

the kernels are different, but they are configured in similar ways.

**Comment by clee604** [comment]
[root@WCV3-Balcony:~]# lsmod
cdc_ether 4493 0 - Live 0xc16e8000
usbnet 23696 1 cdc_ether, Live 0xc16d9000
sensor_gc2053_t31 13463 0 - Live 0xc0681000 (O)
rtl8189ftv 956456 0 - Live 0xc04f4000 (O)
avpu 14329 2 - Live 0xc03aa000 (O)
ch34x 14827 0 - Live 0xc039a000
speaker_ctl 3040 0 - Live 0xc038e000 (O)
sample_pwm_hal 3784 1 - Live 0xc0387000 (O)
sample_pwm_core 3201 1 - Live 0xc0380000 (O)
mmc_detect_test 616 0 - Live 0xc0379000 (O)
sinfo 16620 0 - Live 0xc036f000 (O)
audio 73342 2 - Live 0xc034f000 (O)
exfat 106704 0 - Live 0xc0312000 (O)
tx_isp_t31 986515 1 sensor_gc2053_t31, Live 0xc01d8000 (O)
v4l2loopback 22385 2 - Live 0xc00b2000 (O)

**Comment by gtxaspec** [comment]
Ok, thanks for the info, just trying to rule out stuff thats wrong.  The next step is to make sure your USB port on the v2 actually works physically.  Do you have a USB drive you can plug in?

on the v2, ssh in, plug the flash drive in, wait a few seconds, then run `dmesg`

you should see something like:

```
[   35.656204] usb 1-1: new high-speed USB device number 2 using dwc2
[   36.161655] usb 1-1: device v090c p1000 is not supported
```

**Comment by clee604** [comment]
don't see anything after plugging in the usb flash drive, going to do the same on the v3 to make sure i'm doing it properly

edit: the V3 shows the two lines above, but the V2 does not. Is this a physical hardware limitation (which means v2 can't use ethernet) or is it something in the kernel/drivers with the usb otg?

**Comment by gtxaspec** [comment]
just to verify, you are plugging your ethernet adapters into the big USB type a port on the V2, or the micro usb port?

**Reply by clee604** [reply]
Micro usb port using a OTG with power cable, i've used same cable for both V2 and V3 for my testing

**Comment by gtxaspec** [comment]
gotcha.  The V2 only supports USB devices through the big type-a port.  micro usb otg doesn't work on the V2.

**Reply by clee604** [reply]
ahhh, so I should be plugging the usb devices into the USB A port all this time??

**Comment by gtxaspec** [comment]
right.  micro usb on the v2, is used **only** for power.

**Reply by clee604** [reply]
testing it again now.. and i had painstakingly sanded down my microusb otg cable plug head to fit in the damn V2 microusb port

**Reply by gtxaspec** [reply]
I did the same thing when I first started testing on my V2's...lol.  I still use the same shaved otg adapter on my v3

**Reply by clee604** [reply]
so lsusb now detects the ethernet adapter as well, but lsmod seems to crash the ssh session. I can't tell if the V2 is actually connected via ethernet or still the old wifi connection (the V3 actually had a different IP address when using the lan adapter and the hostname showed up in my router leases so i knew it was on ethernet)

**Reply by gtxaspec** [reply]
run `ifconfig` and check the mac address of wlan0, it should be different than whats printed on the camera.

**Reply by clee604** [reply]
looks like it's still running on wifi, same MAC.. i noticed before yesterday's update when i had the ethernet flag to true, wifi would be completely disabled (ie. wyze app wouldn't work at all), but now even with the ethernet flag set to true, wifi seems to still be active

**Reply by gtxaspec** [reply]
let me test on my V2 and see whats going on, stand by

**Reply by clee604** [reply]
i do see a Segmentation fault after usbnet.ko in the wz_user.log now, this wasn't there in the earlier wz_user.log i posted earlier in the thread
edit: looks like some errors and trace in the dmesg now too

**Reply by gtxaspec** [reply]
I just pushed a new version, that fixes some things with the v2.  not sure if it will fix your issue, but worth a shot.

download it, and do a fresh installation to your memory card. make sure you flash the demo.bin too.  let me know how it works out. ( don't use upgrade-run.sh to upgade at the moment on v2 )

**Reply by gtxaspec** [reply]
I tested the latest version with asix, r8152, and cdc_ether, and its working here for me.  just FYI

**Reply by clee604** [reply]
i've been using the v2_install script every time to produce the firmware (demo.bin) and update it everytime, is this necessary (ie. the hold reset to force load new firmware)? Or can i just do a fresh sd_root folder copy and transfer over my config and ssh keys?

**Comment by gtxaspec** [comment]
if you flashed the demo.bin for the 06-16 release, then you don't have to reflash, you can copy the fresh sd_root folder copy and transfer over your and ssh keys. I would use a fresh config just in case.

but if you used an older demo.bin, flash it once and then you won't have to in the future.

**Comment by clee604** [comment]
Alright, with the new fixes, and plugging the ethernet adapter into the big usb A port I think it is working!

Many thanks for your help, i'm going to go to bed and test it more tomorrow. Deeply appreciate your work on this!

**Reply by gtxaspec** [reply]
you got it! enjoy!!

**Reply by clee604** [reply]
Off hand do you know what the max parameters are for the RTSP feeds for V2 and V3 devices? For example it seems framerate is capped at 15 for V2 and 20 for V3. Are there bitrate and other quality caps that you know of? Now that its hardwired, wifi congestion/bandwidth isn't an issue so I'd like to see where the thresholds are.

edit: Also is there a recommended shutdown/power off procedure? I noticed the sd card got corrupted once (i'm thinking it was writing and i unplugged it). Right now I use the wyze app to 'eject sd card' before powering off but not sure if that works

**Comment by clee604** [comment, marked as answer]
So i tested bit further today and everything seems to be working now. I will mark this as answered. 

For future googlers (and maybe would worthwhile to put in the readme documentation), Wyze V2 cameras do not use an OTG cable like V3 cameras, use the big USB A port in the back for devices.

**Comment by clee604** [comment]
I'm not sure if this is related to the wz_mini_hacks or ethernet, but my V2 is hammering my pihole with DNS requests every 5 seconds.. does yours do that?

Every 5 seconds, DNS request to:
*.credentials.iot.us-west-2.amazonaws.com

edit: doesn't seem to happen on the V3, just the V2. Other thing I noticed is the V3 updates the IP address in the Device info when ethernet is connected, the V2 does not and keeps the wifi IP address shown

**Reply by gtxaspec** [reply]
thats a bug of the factory firmware (dns) and app

**Reply by clee604** [reply]
guess there's nothing we can do but accept it or use V3s? I could block the requests but I rather they just don't create unnecessary traffic.

It seems to continue even when I turn the camera "off" in the wyze app. The RTSP feed through wz_mini continues to work so I would hoping that would stop the wyze DNS calls but nope....

**Reply by gtxaspec** [reply]
i've heard other say rebooting the camera may fix it

**Reply by clee604** [reply]
Tried that and power cycling unfortunately. I'll see if it happens without ethernet tomorrow.

I suspect it may also have something to do with this camera having old legacy cam lite

**Reply by clee604** [reply]
I read that possibly removing the device and readding it to the wyze account could work.

What would be the best order to do this? Flash back the regular demo.bin firmware first? Or I can just remove the sd card and remove from wyze app to re-add?

**Reply by gtxaspec** [reply]
you can keep the sdcard in there, just use the app to remove and re add

**Reply by clee604** [reply]
even if it's configured and plugged into ethernet?

**Reply by gtxaspec** [reply]
actually, remove the card, do the setup on the app, then put the card back on.

**Reply by clee604** [reply]
i actually managed to complete the setup with the card and ethernet still plugged in.. but unfortunately it didn't fix the DNS issue.. it's been pretty frustrating as I should have all motion detection and whatnot disabled....

**Reply by gtxaspec** [reply]
I can try to analyze your system logs to see what is causing them, not happening on any of my cameras... we are not sure why it's happening, users have reported the problem on both V2 and V3 and other models too.  

The logs do also contain some security stuff related to your account.  If that's ok with you sharing for debugging I can show you how to capture the logs.

**Reply by clee604** [reply]
So I've disabled ethernet and went back to wifi and the DNS requests stopped. It did still request on device boot a few AWS servers but not the constant every 5 seconds like it did on ethernet. 

Any ideas? Is there something when ethernet is enabled that intercepts this call to *.credentials.iot.us-west-2.amazonaws.com? (which I assume has buggy wyze code when it doesn't get the right response it slams requests every 5 seconds)

I'm happy to provide some logs, not sure what kind of security stuff should would be included, emails? Passwords? IP addresses?

**Reply by gtxaspec** [reply]
try switching back to ethernet, once the dns stuff happens, run `pkill udhcpc` and see if that stops it.  Somethings the camera runs two dhcp clients.  Other than that, no clue, ethernet is just ethernet, the camera doesn't even know the wifi isn't online.  The stuff in the logs are like encryption keys that can be used to access your account, for some reason they are present in the logs.

**Reply by clee604** [reply]
pkill udhcpc did not stop it. It definitely seems to happen after the switch to ethernet.. as i see the original wifi hostname show up in pihole first.

doesn't happen with the V3 using same adapter..  i'll try another adapter with the V2 but I'm not sure what else I switch around at this point.

**Reply by gtxaspec** [reply]
is it a bunch of amazonaws hosts? or a single one?  the one i see in the code is `c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com`

**Reply by clee604** [reply]
It is that exact host, I didn't know if first part was unique or sensitive (considering it had word 'credentials') so i starred it out.

**Reply by gtxaspec** [reply]
when you see the dns flood, try:
`pkill kvsWebrtcClientMaster`  
and see if it stops

**Reply by clee604** [reply]
thanks, will reconnect ethernet and test now. Is there a reboot command that I can use in SSH? 'reboot now' doesn't work but my linux knowledge is limited

**Reply by gtxaspec** [reply]
just `reboot`

**Reply by clee604** [reply]
unfortuantely did not stop the dns calls every 5 seconds to that iot server

**Reply by clee604** [reply]
I noticed that pkill kvsWebrtcClientMaster kills the service only for a moment, then pgrep kvsWebrtcClientMaster shows that it is back...

**Reply by gtxaspec** [reply]
you are right, it wil try to relaunch it forever, and if its deleted it re-downloads it.  I will find an alternate solution.

**Reply by gtxaspec** [reply]
one last thing if you want to try:
`touch /tmp/test_hosts;echo -e "127.0.0.1 localhost\n54.185.42.140 c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com" > /tmp/test_hosts; mount --bind /tmp/test_hosts /etc/hosts`

**Reply by clee604** [reply]
that seems to have stopped the dns requests to my pihole! I'll see how long it lasts later.. need to reenable wifi as I'm not sure if my tests for the new wifi driver were reversed so i want to retest it

**Reply by gtxaspec** [reply]
great keep us updated!

**Reply by clee604** [reply]
phew, didn't mess up my iperf3 numbers in the other thread. I will keep the V2 connected to ethernet overnight and see if the dns requests come back. Do you know if any consequences for blacklisting that domain in the hosts file?

**Reply by gtxaspec** [reply]
no issues until the ip changes on that Amazon server, it's technically not blacklisted, we force the domain name to the local hosts file io address so the request goes to the system, not the dns server.

**Reply by clee604** [reply]
if this works, is there a way to have this persist reboots? Would I put this in the custom scripts section? Thanks for all the help!

**Reply by clee604** [reply]
update: left camera v2 on overnight and so far still no DNS floods!

**Comment by clee604** [comment]
@gtxaspec I think there may be a regression in the newest version. I am unable to connect using ethernet again on V2.

When on wifi mode but with ethernet connected lsusb shows the adapter so it is connected
`Bus 001 Device 004: ID 0bda:8152
Bus 001 Device 001: ID 1d6b:0002`

With ethernet =true, wz_user.log shows
+ first_run_check
+ '[' -e /opt/wz_mini/tmp/.wz_user_firstrun ]
+ wait_wlan
+ true
+ grep 'inet addr'
+ ifconfig wlan0
ifconfig: wlan0: error fetching interface information: Device not found
+ grep 'inet addr'
+ ifconfig wlan0
ifconfig: wlan0: error fetching interface information: Device not found
+ echo ' wlan0 not ready yet...'
 wlan0 not ready yet...
+ sleep 5
+ true
+ grep 'inet addr'
+ ifconfig wlan0
+ grep 'inet addr'
+ ifconfig wlan0
+ echo ' wlan0 not ready yet...'
 wlan0 not ready yet...
+ sleep 5
+ true
+ grep 'inet addr'
+ ifconfig wlan0
+ grep 'inet addr'
+ ifconfig wlan0
+ echo ' wlan0 not ready yet...'
 wlan0 not ready yet...
+ sleep 5
+ true
+ grep 'inet addr'
+ ifconfig wlan0
+ grep 'inet addr'
+ ifconfig wlan0
+ echo ' wlan0 not ready yet...'
 wlan0 not ready yet...
+ sleep 5
+ true
+ grep 'inet addr'
+ ifconfig wlan0
+ grep 'inet addr'
+ ifconfig wlan0
+ echo ' wlan0 not ready yet...'
 wlan0 not ready yet...
+ sleep 5
+ true
+ grep 'inet addr'
+ ifconfig wlan0
+ grep 'inet addr'
+ ifconfig wlan0
+ echo ' wlan0 not ready yet...'
 wlan0 not ready yet...
+ sleep 5

**Reply by gtxaspec** [reply]
let me test on v2 again

**Reply by gtxaspec** [reply]
works on my end, try generating a new demo.bin and try again...

**Reply by clee604** [reply]
i tried switching to a cdc_ether adapter and it got further but weird error at the end. I'll try to blow everything up, reformat the sd card and reinstall demo.bin this evening to see if it fixes it.

wz_user.log
+ dmesg_log
+ DMESG_LOG=/opt/wz_mini/log/dmesg
+ '[[' -e /opt/wz_mini/log/dmesg.log '||' -L /opt/wz_mini/log/dmesg.log ]]
+ i=0
+ '[[' -e /opt/wz_mini/log/dmesg.log.0 '||' -L /opt/wz_mini/log/dmesg.log.0 ]]
+ mv /opt/wz_mini/log/dmesg.log /opt/wz_mini/log/dmesg.log.0
+ DMESG_LOG=/opt/wz_mini/log/dmesg
+ touch -- /opt/wz_mini/log/dmesg.log
+ dmesg
+ trim_logs
+ echo 'Deleting logs older than 5 boots...'
Deleting logs older than 5 boots...
+ read file
+ find /opt/wz_mini/log -name '*log*'
log level:500
+ '[' /opt/wz_mini/log -gt 5 ]
sh: /opt/wz_mini/log: out of range
+ read file
+ '[' /opt/wz_mini/log/wz_init.log -gt 5 ]
sh: /opt/wz_mini/log/wz_init.log: out of range
+ read file
+ '[' /opt/wz_mini/log/wz_post.log -gt 5 ]
sh: /opt/wz_mini/log/wz_post.log: out of range
+ read file
+ '[' /opt/wz_mini/log/wz_user.log -gt 5 ]
sh: /opt/wz_mini/log/wz_user.log: out of range
+ read file
+ '[' 0 -gt 5 ]
+ read file
+ '[' 0 -gt 5 ]
+ read file
+ '[' 0 -gt 5 ]
+ read file
+ '[' 1 -gt 5 ]
+ read file
+ '[' 1 -gt 5 ]
+ read file
+ '[' 1 -gt 5 ]
+ read file
+ '[' /opt/wz_mini/log/dmesg.log -gt 5 ]
sh: /opt/wz_mini/log/dmesg.log: out of range
+ read file
+ '[' 2 -gt 5 ]
+ read file
+ '[' 2 -gt 5 ]
+ read file
+ '[' 2 -gt 5 ]
+ read file
+ '[' 0 -gt 5 ]
+ read file
+ sync
+ echo 3
+ '[' -f  ]
+ echo 'custom script not found'
custom script not found
+ echo 'wz_user.sh done'
+ event=n
+ directory=/tmp
+ file=tmp.cAniDe
+ date
Fri Jun 24 17:44:14 UTC 2022
+ '[[' tmp.cAniDe '==' img ]]

dmesg.log
[   39.982418] cdc_ether 1-1:2.0 eth0: register 'cdc_ether' at usb-dwc2-1, CDC Ethernet Device, 68:e4:3b:30:61:a9
[   40.057616] usbcore: registered new interface driver cdc_ether
[   40.116230] RTW: ERROR Free disconnecting network of scanned_queue failed due to pwlan == NULL
[   40.116230] 
[   40.134347] cfg80211: Calling CRDA to update world regulatory domain
[   44.257412] RTW: nolinked power save enter

**Comment by clee604** [comment]
No dice...  fresh nuked install and get stuck at what looks like the ethernet switch over. It looks like it's not disconnecting the wifi properly?

Logs is same error as before.

dmesg.log
[   66.434509] cdc_ether 1-1:2.0 eth0: register 'cdc_ether' at usb-dwc2-1, CDC Ethernet Device, 68:e4:3b:30:61:a9
[   66.502116] usbcore: registered new interface driver cdc_ether
[   66.603905] RTW: ERROR Free disconnecting network of scanned_queue failed due to pwlan == NULL
[   66.603905] 
[   66.633382] cfg80211: Calling CRDA to update world regulatory domain
[   71.177260] RTW: nolinked power save enter

wz_user.log
+ dmesg_log
+ DMESG_LOG=/opt/wz_mini/log/dmesg
+ '[[' -e /opt/wz_mini/log/dmesg.log '||' -L /opt/wz_mini/log/dmesg.log ]]
+ touch -- /opt/wz_mini/log/dmesg.log
+ dmesg
+ trim_logs
+ echo 'Deleting logs older than 5 boots...'
Deleting logs older than 5 boots...
+ + find /opt/wz_mini/log -name '*log*'
read file
+ '[' /opt/wz_mini/log -gt 5 ]
sh: /opt/wz_mini/log: out of range
+ read file
+ '[' /opt/wz_mini/log/wz_init.log -gt 5 ]
sh: /opt/wz_mini/log/wz_init.log: out of range
+ read file
+ '[' /opt/wz_mini/log/wz_post.log -gt 5 ]
sh: /opt/wz_mini/log/wz_post.log: out of range
+ read file
+ '[' /opt/wz_mini/log/wz_user.log -gt 5 ]
sh: /opt/wz_mini/log/wz_user.log: out of range
+ read file
+ '[' /opt/wz_mini/log/dmesg.log -gt 5 ]
sh: /opt/wz_mini/log/dmesg.log: out of range
+ read file
+ sync
+ echo 3
log level:500
+ '[' -f  ]
+ echo 'custom script not found'
custom script not found
+ echo 'wz_user.sh done'
+ event=n
+ directory=/tmp
+ file=tmp.pipjLJ
+ date
Sun Jun 26 16:23:40 UTC 2022
+ '[[' tmp.pipjLJ '==' img ]]
+ event=n
+ directory=/tmp
+ file=aws_iot
+ date
Sun Jun 26 16:27:12 UTC 2022
+ '[[' aws_iot '==' img ]]
+ event=n
+ directory=/tmp
+ file=aws_iot
+ date
Sun Jun 26 16:27:12 UTC 2022
+ '[[' aws_iot '==' img ]]

**Comment by gtxaspec** [comment]
can you verify that wifi works though?

**Comment by clee604** [comment]
Wifi and everything works fine with ethernet disabled in configuration. For the time being I'm just running wifi only

**Comment by clee604** [comment]
So either there's some kind of regression for V2s in the new versions or my V2 is dying. The video feed in the Wyze app / RTSP feed goes down after awhile. I can still SSH into the camera, but cannot access the video feed until I reboot it.

I checked the wz_user.log and didn't see anything, are there other logs I can look at to see what crashed the video stream?

**Comment by clee604** [comment]
@gtxaspec I didn't want to deviate the other V3 RTSP thread, but your post about testing older version led me to drop back my V2 to test older releases.
This release, my ethernet works
https://github.com/gtxaspec/wz_mini_hacks/tree/66031b41f53ad5578c31c2b9a56e52211fdac85d

The release you posted in the other thread, ethernet doesn't work
https://github.com/gtxaspec/wz_mini_hacks/tree/cf02d1aeb96fafde1b19d5eb25a22b705e536df5

**Comment by gtxaspec** [comment]
@clee604 thanks for the info.  which driver is your Ethernet adapter?

**Reply by clee604** [reply]
@gtxaspec I've tested both r8152 and cdc_ether on the V2. 

I also have an asix adapter but it is currently being used by a V3 (which has the rtsp issues, but the stream is stable enough that we still use it to monitor a baby play area so I don't take it offline to test unless baby is sleeping)

**Comment by clee604** [comment]
i reflashed the new version and installed the new wifi driver from the other thread. Wifi seems to be working fine (ethernet still didn't work), I didn't have artifacts on the V2 so can't test that and I'll leave it on to see the video feed ends up crashing like it did before.
